project/src/main/puzzle/piece/piece-mover.gd

```diff
@@ -120,9 +151,23 @@ func apply_move_input(piece: ActivePiece) -> void:
 		input.set_left_das_active()
 	if input.is_right_pressed() and not piece.can_move_to(piece.pos + Vector2.RIGHT, piece.orientation):
 		input.set_right_das_active()
+	
+	return movement_signal
 
 
-## Move piece once per frame to allow pieces to slide into nooks during 20G.
+## Move piece during 20G to allow pieces to slide into nooks.
 func attempt_mid_drop_movement(piece: ActivePiece) -> void:
-	if _horizontal_movement_count == 0:
-		apply_move_input(piece)
+	if _normal_horizontal_movement_count > 0:
+		# piece moved normally this frame, do not apply mid-drop movement
+		return
+	
+	# attempt a single mid-drop movement
+	var movement_signal := apply_move_input(piece, true)
+	
+	if movement_signal != "" and (input.is_left_das_active() or input.is_right_das_active()):
```

Comment:
Couldn't this check for the movement signal's value of `das_moved_left` to be a little more concise?

Reply:
Updated code to check the movement signal instead of checking `input` directly